phoenix-core-client/src/main/java/org/apache/phoenix/schema/PMetaData.java

```diff
@@ -29,6 +29,7 @@ public static interface Pruner {
     }
     public int size();
     public PTableRef getTableRef(PTableKey key) throws TableNotFoundException;
+    public PTableRef getTableRefOptimized(PTableKey key);
```

Comment:
> rather it is in selection of right method (the new code PhoenixConnection)

But the selection of different method is all about avoiding creating TNFE object so, the optimization is about not throwing and creating TNFE. Right?

Reply:
I think we are both saying the same thing, the optimization is about method selection, but my point is that the new method is just a variant that facilitates this optimization but not directly addressing the optimization itself. The optimization comes from the use case, e.g., say you have a use case that calls the same method but converts the `null` to TNFE (basically what the existing getTableRef does, but now this is the application logic), now this new application code won't be optimized just because it is using this new method. Not the best argument possible, but I hope it conveys the meaning.